ampd/src/stellar/verifier.rs

```diff
@@ -0,0 +1,224 @@
+use std::str::FromStr;
+
+use axelar_wasm_std::voting::Vote;
+use stellar_xdr::curr::{ContractEventBody, ScAddress, ScSymbol, ScVal, StringM};
+
+use crate::handlers::stellar_verify_msg::Message;
+use crate::stellar::http_client::TxResponse;
+
+const TOPIC_CALLED: &str = "called";
+
+impl PartialEq<&ContractEventBody> for &Message {
+    fn eq(&self, event: &&ContractEventBody) -> bool {
+        let ContractEventBody::V0(body) = event;
+
+        if body.topics.len() != 3 {
+            return false;
+        }
+
+        let [symbol, source_address, payload_hash] = &body.topics[..] else {
+            return false;
+        };
+
+        let expected_topic: ScVal =
+            ScSymbol(StringM::from_str(TOPIC_CALLED).expect("must convert str to ScSymbol")).into();
+
+        symbol == &expected_topic
+            && (source_address == &ScVal::Address(self.source_address.clone()))
+            && (payload_hash == &ScVal::Bytes(self.payload_hash.clone()))
+            && match &body.data {
+                ScVal::Vec(Some(data)) if data.len() == 3 => {
+                    let [dest_chain, dest_address, _] = &data[..] else {
+                        return false;
+                    };
+                    (ScVal::String(self.destination_chain.clone()) == *dest_chain)
+                        && (ScVal::String(self.destination_address.clone()) == *dest_address)
+                }
+                _ => false,
```

Comment:
can you extract that into a separate variable? This is a bit too condensed

Reply:
updated